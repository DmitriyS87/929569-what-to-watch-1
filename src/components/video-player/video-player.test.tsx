import * as React from 'react';
import * as renderer from 'react-test-renderer';
import VideoPlayer from './video-player';
import mockData from '../../mocks/test-mocks.js';

it(`VideoPlayer correctly renders after relunch`, () => {
  const mockFunction = jest.fn();
  const tree = renderer
    .create(<VideoPlayer
      poster={mockData.film.previewImg.src}
      movies={mockData.film.previewMovie}
      title={mockData.film.title}
      onMouseLeave={mockFunction}
    />, {
        createNodeMock: (element) => {
          if (element.type === `video`) {
            return {
              focused: true,
              autoplay: true,
              controls: true,
              mute: true,
              onMouseLeave: mockFunction,
              onError: mockFunction,
            };
          }
          return null;
        }
      }
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
