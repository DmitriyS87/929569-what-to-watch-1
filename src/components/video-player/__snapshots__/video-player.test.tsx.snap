// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer correctly renders after relunch 1`] = `
<video
  className="player__video"
  poster="img/aviator.jpg"
>
  <source
    src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
    type="video/webm"
  />
  <source
    src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
    type="video/mp4"
  />
  К сожалению, ваш браузер не поддерживает HTML5 Video.
</video>
`;
